src/node/serialization.cc

```diff
@@ -147,6 +147,8 @@ struct JSONNode {
   std::vector<std::string> keys;
   /*! \brief values of a map or array. */
   std::vector<size_t> data;
+  /*! \brief field member dependency. */
+  std::vector<size_t> fields;
```

Comment:
Add NOTE, this is an auxiliary data structure for loading, won't be serialized to json